solutions/devpsrint-michael-douglas-1/GitHubApp/Screens/List/ListView.swift

```diff
@@ -10,6 +10,7 @@ import UIKit
 final class ListView: UIView {
 
     // MARK: - View Properties
+    var goToDetailsViewController: ((Repository) -> Void)?
```

Comment:
Para tornar o seu código mais testável e também desacoplado, considere criar 2 protocolos.

1º - O primeiro protocolo vai ser para a sua `ListView`, ele pode se chamar `ListViewProtocol` e vai ser responsável por fazer a comunicação entre a sua Controller e a View, assim, você terá que declarar o método de `func updateView` nela e todos os outros métodos ou propriedades que a sua controller chama na View. Depois que fizer isso, não se esqueça de ir na controller e implementar a `ListView` com o protocolo e fazer a injeção dela como dependência.

2º - O segundo protocolo será um Delegate, e será responsável por fazer a comunicação entre a sua View e a Controller, assim, você terá que criar nele o método de `func didSelect(_ repository: Repository)`, que será chamado na sua controller e assim você seguirá o fluxo normal das coisas.

Dessa forma, você torna o seu código testável, por que depois você conseguirá facilmente mockar os protocolos e injetar seus Dublês para fazer os testes =D
```suggestion
    weak var delegate: ListViewDelegate?
```

Reply:
Ponto 2 foi corrigido. O ponto 1 acredito que foram outras pessoas do time que implementaram e talvez podemos implementar juntos no encontro sync. Se achar melhor posso implementar aqui também sem problemas.

Reply:
Nice, eu acho uma boa!!! Podemos implementar isso no nosso encontro sim, para dar visibilidade para todos de como podemos fazer utilizando protocolos 🚀